mono/metadata/w32file-unix.c

```diff
@@ -2359,7 +2365,17 @@ CopyFile (const gunichar2 *name, const gunichar2 *dest_name, gboolean fail_if_ex
 		
 		return(FALSE);
 	}
-	
+
+#if HAVE_COPYFILE_H
+	ret = copyfile (utf8_src, utf8_dest, NULL, COPYFILE_ALL | COPYFILE_CLONE | (fail_if_exists ? COPYFILE_EXCL : 0));
```

Comment:
This won't work because COPYFILE_CLONE implies COPYFILE_EXCL (https://opensource.apple.com/source/copyfile/copyfile-146.30.2/copyfile.c.auto.html). One way to solve that is to fallback on `if (ret != 0 && errno == EEXIST && ! fail_if_exists)` to non-cloning variant. I wanted to check why COPYFILE_UNLINK didn't work, but the build results are gone now :-/

Reply:
Nvm, I see it now - same file name for src and dest is broken for COPYFILE_UNLINK.